thinc/backends/ops.py

```diff
@@ -680,6 +682,167 @@ def backprop_relu(
         dY *= Y > 0
         return dY
 
+    def relu_n(
```

Comment:
I was thinking about this, but I was wondering whether its better to do it in the next PR where I'll try to improve the consistency of the activation function `ops` + add a generic `Dense(dim=256, activation="gelu")` wrapper layer.

Reply:
Since this is a new function, I think it makes sense to do it right the first time.

Reply:
Then I guess `relu_n` can just become a `ops.clip(min_val, max_val)` with an adjoint `dY * self.clip(X, min_val, max_val) != 0`

Reply:
I was just also not sure where should this stop because it can be even more general with a slope and an offset.

Reply:
Or like if you look at the `hard_swish` functions they are the same but then times x. 

like the very general version I guess would be like `(x or 1) *  self.xp.clip(x * slope + offset, min_val, max_val)`.

Reply:
For hard sigmoids these are the ones ive came across

Hard-sigmoid:

https://arxiv.org/pdf/1603.00391.pdf

`clip(x * 0.25 + 0.5, 0, 1)`

https://arxiv.org/pdf/1511.00363v3.pdf

`clip(x * 0.5 + 0.5, 0, 1)`


https://pytorch.org/docs/stable/generated/torch.nn.Hardsigmoid.html

`x * 0.166667 + 0.5, x[x <= -3] = 0, x[x >=3] = 1`

Reply:
There are many different abstraction levels. My main motivation was to have a function that is similar to what is offered in other libraries.

But it was only a minor suggestion anyway, feel free to ignore.

Reply:
I agree that it would be nice, but I was just not sure what is that other libraries offer for these hard functions. I looked at the hard sigmoid in Keras:

https://github.com/keras-team/keras/blob/2c48a3b38b6b6139be2da501982fd2f61d7d48fe/keras/backend.py#L5196

and Pytorch

https://pytorch.org/docs/stable/generated/torch.nn.Hardsigmoid.html

but they didn't seem to be the same.  For the MobileNetV3 application its like the Pytorch version then: 

https://github.com/keras-team/keras/blob/v2.7.0/keras/applications/mobilenet_v3.py#L455

So would you think it makes sense to implement a generic `ops.clipped_linear` with

`clip(x * slope + 0.5, min_val, max_val)`. This can do `relu`, `relu_n`, `hard_tanh` and `hard_sigmoid`. And this then comes with a generic `ops.backprop_clipped_linear`?

Reply:
I've pushed the generic `clipped_linear`, let me know what do you think! `test_clipped_linear` shows how it can implement `relu6` and `hard_sigmoid`. `hard_tanh` would be `ops.clipped_linear(min_val=-1, max_val=1)`, but it could do other stuff too.

Reply:
Nice 👍🏻🎉!